skimage/measure/_regionprops_utils.py

```diff
@@ -174,13 +181,53 @@ def euler_number(image, connectivity=None):
     # XF has values in the 0-255 range in 3D, and in the 0-15 range in 2D,
     # with one unique value for each binary configuration of the
     # 27-voxel cube in 3D / 8-pixel square in 2D, up to symmetries
-    XF = ndi.convolve(image, config, mode='constant', cval=0)
-    h = np.bincount(XF.ravel(), minlength=bins)
+    XF = convolve_along_axes(image, config, image_axes, mode='constant', cval=0)
+    XF_flat = flatten_along_axes(XF, axes=image_axes)
+
+    bin_edges = np.arange(0, bins+0.1, 1)
+    h = histogram_along_axes(XF_flat, bins=bin_edges, axes=-1 if coord_dims else None)
```

Comment:
I think the following is equivalent and avoids the need for the dependency. It also runs a bit faster on the case I tried it on (see example below)

```suggestion
    h = np.empty((XF_flat.shape[0], bins), dtype=int)
    for i, x in enumerate(XF_flat):
        h[i, :] = np.bincount(x, minlength=bins)
```

I tried this using a simple benchmark:
```Python
import numpy as np
many_samples = np.ones((256, 512, 512), dtype=bool)
many_samples[:, ::16, :] = 0
many_samples[:, :, ::16] = 0
euler_number(many_samples, axes=[1, 2]
```
and the duration after the suggested change was 1.05 s vs. 1.88 s prior to the change.